### PR TITLE
Travis dependent builds

### PR DESCRIPTION
**Pull request by diegogrz**
- [x] @joeferner
- [x] @diegogrz
- [x] @mwizeman @sfeng88
- [ ] @joeybrk372 @rygim @jharwig @EvanOxfeld

Testing Instructions:
You can download the script from travis/travis-request.sh and follow the instructions in travis/README.md to use the script to trigger a Travis build from a workstation.

CHANGELOG
Added: travis/travis-request.sh helper script to generate Travis build requests
Documentation: travis/README.md Helper script documentation

### CODE REVIEW

**Comment by kunklejr** [comment]
The PR itself looks great, although I'm not excited about the open source repo referencing a private repo. Is there any way around that?

File: .travis.yml [hunk 1]
```diff
@@ -9,6 +9,8 @@ after_success:
   - test $TRAVIS_REPO_SLUG == "v5analytics/visallo" && test $TRAVIS_BRANCH == "master" || test 'echo $TRAVIS_BRANCH | grep -Eq /^release-.*$/' && test $TRAVIS_PULL_REQUEST == "false" && echo "<settings><servers><server><id>sonatype-nexus-snapshots</id><username>\${env.DEPLOY_USERNAME}</username><password>\${env.DEPLOY_PASSWORD}</password></server></servers></settings>" > ~/settings.xml
```
Comment (mwizeman) [inline .travis.yml]: We should update the repo organization name here from v5analytics to visallo

**Comment by diegogrz** [comment]
@kunklejr We can use an environment variable instead: TRAVIS_DOWNSTREAM_REPO. We can obfuscate the private repo name, but can't obfuscate the fact that there is a "downstream" component.

**Comment by kunklejr** [comment]
LGTM

**Pull request merged**